lib/generator.js

```diff
@@ -283,20 +291,27 @@
         }
         
         if (deferred.promise.isPending()) {
-            readFile(resolve(__dirname, path), {encoding: "utf8"}, function (err, data) {
-                var id;
-
-                if (err) {
-                    deferred.reject(err);
-                } else {
-                    data = "var params = " + paramsString + ";\n" + data;
-                    id = self._photoshop.sendCommand(data);
-                    self._messageDeferreds[id] = deferred;
-
-                    deferred.promise.finally(function () {
-                        delete self._messageDeferreds[id];
-                    });
-                }
+            var id;
+            if (self._sendJSXCache.hasOwnProperty(path) && self._sendJSXCache[path] !== "") {
+                var data = "var params = " + paramsString + ";\n" + data;
+                id = self._photoshop.sendCommand(data);
```

Comment:
Does this even construct the right string? where is the value in `self._sendJSXCache[path]` actually put into `data`? Am I blind?